reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java

```diff
@@ -681,4 +693,111 @@ static OnSetup empty() {
 			Subscription.class,
 			"outboundSubscription");
 
+	static final class DisposedChannel extends AbstractChannel {
+
+		final DefaultChannelConfig config;
+		final SocketAddress localAddress;
+		final ChannelMetadata metadata;
+		final SocketAddress remoteAddress;
+
+		DisposedChannel(Channel copy) {
+			super(null);
+			this.metadata = copy.metadata();
+			this.config = new DefaultChannelConfig(this);
+			this.localAddress = copy.localAddress();
+			this.remoteAddress = copy.remoteAddress();
+		}
+
+		@Override
+		public ChannelFuture closeFuture() {
+			return newSucceededFuture();
+		}
+
+		@Override
+		public ChannelConfig config() {
+			return config;
+		}
+
+		@Override
+		protected void doBeginRead() {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		protected void doBind(SocketAddress socketAddress) {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		protected void doClose() {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		protected void doDisconnect() {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		protected void doWrite(ChannelOutboundBuffer channelOutboundBuffer) {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		public boolean isActive() {
+			return false;
+		}
+
+		@Override
+		protected boolean isCompatible(EventLoop eventLoop) {
+			return false;
+		}
+
+		@Override
+		public boolean isOpen() {
+			return false;
+		}
+
+		@Override
+		protected SocketAddress localAddress0() {
+			return localAddress;
+		}
+
+		@Override
+		public ChannelMetadata metadata() {
+			return metadata;
+		}
+
+		@Override
+		protected AbstractUnsafe newUnsafe() {
+			return new DisposedChannelUnsafe();
+		}
+
+		@Override
+		protected SocketAddress remoteAddress0() {
+			return remoteAddress;
+		}
+
+		final class DisposedChannelUnsafe extends AbstractUnsafe {
+
+			@Override
+			public void connect(SocketAddress remoteAddress, SocketAddress localAddress, ChannelPromise promise) {
+				promise.setFailure(new UnsupportedOperationException());
+			}
+		}
+	}
+
+	static final class DisposedConnection implements Connection {
+
+		final Channel channel;
+
+		DisposedConnection(Channel copy) {
```

Comment:
I struggle with the name a bit. It's not really a copy as the name would imply. The behaviour you expect is to copy so you're treating a name as a verb, which is confusing. Am I getting it wrong? Perhaps `Channel actual` would be indicative that we're working with something that wraps the actual `Channel`. Then the same comment would apply to the argument of `DisposedChannel`'s constructor.

Reply:
change applied